### PR TITLE
Comment, Bookmark & Like

### PR DESCRIPTION
**Pull request by vishalg8454**
This PR adds the functionality of Liking, Bookmarking and Commenting on a Post. 

The redux slices added are - 
- `bookmarkSlice` at src/features

The pages added are -
- `BookmarkPage` at src/pages/bookmark-page

[live preview](https://deploy-preview-5--stalk-me.netlify.app/)

Would love to hear your feedback and suggestions.

image

image

### CODE REVIEW

File: src/components/post/Post.jsx [hunk 1]
```diff
@@ -14,7 +14,16 @@ import { Portal, PortalForModal, EditModal } from "../../components";
 import parse from "html-react-parser";
 import { Comment } from "./Comment";
 import { useSelector, useDispatch } from "react-redux";
-import { deletePost } from "../../features/post/postSlice";
+import {
+  deletePost,
+  likePost,
+  unlikePost,
+  addCommentToPost,
+} from "../../features/post/postSlice";
```
Comment (viru2001) [inline src/components/post/Post.jsx:22]: you can use [Absolute imports](https://dev.to/mr_frontend/absolute-imports-in-create-react-app-3ge8) here

File: src/components/post/Post.jsx [hunk 2]
```diff
@@ -26,15 +35,53 @@ const Post = ({
...
   const { user, token } = useSelector((store) => store.auth);
+  const { posts, status } = useSelector((store) => store.post);
+  const { bookmarks, status: bookmarkStatus } = useSelector(
+    (store) => store.bookmark
+  );
   const [replyText, setReplyText] = useState("");
   const [menuOn, setMenuOn] = useState(false);
   const [editModalOn, setEditModalOn] = useState(false);
+  const [isLiked, setIsLiked] = useState(false);
+  const [isBookmarked, setisBookmarked] = useState(false);
```
Comment (viru2001) [inline src/components/post/Post.jsx:46]: you can use reducer here

**Review by viru2001 — APPROVED** [review]

File: src/components/post/Post.jsx [hunk 3]
```diff
@@ -46,12 +93,42 @@ const Post = ({
...
-            <button className="hover:opacity-75">
-              <FavoriteBorderIcon />
-            </button>
-            <button className="hover:opacity-75">
-              <BookmarkBorderOutlinedIcon />
-            </button>
+            {!isLiked && (
+              <button
+                disabled={status === "loading"}
+                className="hover:opacity-75 disabled:cursor-not-allowed"
```
Comment (Developers-codz) [inline src/components/post/Post.jsx:99]: why are you providing styling in className ??

File: src/features/bookmark/bookmarkSlice.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        {},
+        {
+          headers: {
+            authorization: token,
+          },
+        }
+      );
+      return res.data;
+    } catch (error) {
+      thunkAPI.rejectWithValue(error);
```
Comment (Developers-codz) [inline src/features/bookmark/bookmarkSlice.js:92]: Appreciation: for using thunkAPI and doing proper error handling with it

File: src/pages/home-page/HomePage.jsx [hunk 5]
```diff
@@ -14,11 +14,11 @@ const HomePage = () => {
...
       <NewPost />
       <p className="text-xl mt-2 font-bold">Latest Posts</p>
-      {status === "loading" && (
-        <div className="flex justify-center">
+      {/* {status === "loading" && (
+        <div className="flex justify-center fixed text-center left-0 right-0">
           <Loader />
         </div>
-      )}
+      )} */}
```
Comment (Developers-codz) [inline src/pages/home-page/HomePage.jsx:21]: remove commented code if not needed

**Review by Developers-codz — APPROVED** [review]
UI is nice, code also looks good and readable, you can merge
Found one bug in UI, your search bar is overflowing you can decrease the z-index of it

File: src/components/post/Post.jsx (hunk 3 above)
Comment (vishalg8454) [reply src/components/post/Post.jsx:99]: Yes, I am using Tailwind CSS that's why.

**Pull request merged**